env/mock_env.cc

```diff
@@ -1034,57 +1100,30 @@ Status MockFileSystem::CorruptBuffer(const std::string& fname) {
   iter->second->CorruptBuffer();
   return Status::OK();
 }
-namespace {
-class MockSystemClock : public SystemClockWrapper {
- public:
-  explicit MockSystemClock(const std::shared_ptr<SystemClock>& c)
-      : SystemClockWrapper(c), fake_sleep_micros_(0) {}
-
-  void FakeSleepForMicroseconds(int64_t micros) {
-    fake_sleep_micros_.fetch_add(micros);
-  }
 
-  const char* Name() const override { return "MockSystemClock"; }
+MockEnv::MockEnv(Env* env, const std::shared_ptr<FileSystem>& fs,
```

Comment:
The new TimeElapesSystemClock class looks good to me in general (the name may not be that appropriate......). Can you describe what's the testing failure when using MockSystemClock and why it is resolved in current implementation?

Reply:
The TimeElapseSystemClock was extracted from SpecialEnv.  Feel free to suggest a better name.  SpecialClock sounded nondescript and there is another class already called MockSystemClock.

There were some failures with the "FakeSystemClock" when running under MEM_ENV.  The problem is that the SpecialEnv would do a "Sleep" and defer to the wrapper clock, which in this case was a FakeSystemClock that did a fake sleep.  Tests that were depending on a real sleep would therefore fail.  I believe the failures were typically around the rate limiter tests but my recollection may be off.